msatutil/mair_geoviews.py

```diff
@@ -428,6 +432,7 @@ def read_variables(
                     v[nan_num_samples] = np.nan
                 if i == 0 and option is not None:
                     v = getattr(np, option)(v, axis=nc[var].dimensions.index(option_axis_dim))
+                    assert nc[apply_flag][:].shape == v.shape, f"Flag variable {apply_flag} has shape {nc[apply_flag][:].shape}, but first variable {var} has shape {v.shape}"

```

Comment:
this will throw an error when apply_flag is None

Reply:
Whoops, I had inserted this catch on the wrong line.